grease-cli/src/Grease/Main.hs

```diff
@@ -598,8 +629,19 @@ simulateMacawCfg la bak fm halloc macawCfgConfig archCtx simOpts setupHook mbCfg
   let mdEntryAbsAddr = fmap (segoffToAbsoluteAddr memory) mbCfgAddr
   initArgs_ <- minimalArgShapes bak archCtx mdEntryAbsAddr
   let argNames = fmapFC (Const . getValueName) rNameAssign
+  -- Is there a better way to do this resolution? We need the pc absent any position independent loading
+  -- so we find the segment that corresponds by checking for all segments if segindex->memsegment -> the segment of the target addr
+  let segIndexToSegment = MM.memSegmentIndexMap memory
+  let targetSegment = fst <$> ((\seg -> find (\(_, seg2) -> seg == seg2) (Map.toList segIndexToSegment)) =<< (MM.segoffSegment <$> mbCfgAddr))
+  let mbSegment = (\bin -> find (\phdr -> (Just $ Elf.phdrSegmentIndex phdr) == targetSegment) $ headerPhdrs bin) =<< mcElf macawCfgConfig
+  let mbNewAddr =
+        ( \(seghdr, entAddr) ->
+            (MM.memWordValue $ MM.segoffOffset entAddr) + fromIntegral (Elf.phdrSegmentVirtAddr seghdr)
+        )
+          <$> ((,) <$> mbSegment <*> mbCfgAddr)
+  dwarfedArgs <- loadDwarfpreconditions (simEnableDWARFPreconditions simOpts) argNames initArgs_ macawCfgConfig archCtx mbNewAddr
```

Comment:
...specifically, here would be a good place to control whether to call this function or not. Something like:

```hs
  dwarfedArgs <-
    if simEnableDWARFPreconditions simOpts
      then loadDwarfPreconditions argNames initArgs_ macawCfgConfig archCtx mbNewAddr
      else pure initArgs_
```